lib/sudo-exec/src/monitor.rs

```diff
@@ -97,14 +100,18 @@ impl MonitorClosure {
     /// A signal is self-terminating if `signaler_pid`:
     /// - is the same PID of the command, or
     /// - is in the process group of the command and the command is the leader.
-    fn is_self_terminating(&self, signaler_pid: ProcessId) -> bool {
+    fn is_self_terminating(
```

Comment:
nit: this is pretty much down to personal style but I'm not a fan of associated functions that don't have `Self` in the signature. I would move these into free functions